compiler-cli/src/add.rs

```diff
@@ -36,18 +22,34 @@ pub fn command(to_add: String, dev: bool) -> Result<()> {
             err: Some(e.to_string()),
         })?;
 
-    // Insert the new dep
-    #[allow(clippy::indexing_slicing)]
-    if dev {
-        toml["dev-dependencies"][&to_add] = toml_edit::value(range);
-    } else {
-        toml["dependencies"][&to_add] = toml_edit::value(range);
-    };
+    // Insert the new deps
+    for package_to_add in packages {
+        // Pull the selected version out of the new manifest so we know what it is
+        let version = &manifest
+            .packages
+            .iter()
+            .find(|package| package.name == *package_to_add)
```

Comment:
unless the `manifest` has a smart implementation of `find`, this is an `O(n*m)` operation. But since people won't be adding thousands of packages at a time this shouldn't be a problem

Reply:
Well spotted, but aye it's a tiny amount of data and we run it once, so no problem.